File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int calc(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");

    // Call func to count letters
    int letters = count_letters(text);

    // Call func to count words
    int words = count_words(text);

    // Call func to count sentences
    int sentences = count_sentences(text);

    // calculate the grade
    int c = calc(letters, words, sentences);

    // if grade is between 1 and 15
    if (c >= 1 && c <= 15)
    {
        printf("Grade %i\n", c);
    }
    else if (c < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }

}


int count_letters(string text)
{
    int letters = 0;
    // Loop through string until null character
    for (int i = 0; text[i] != '\0' ; i++)
    {
        // if current c is an alphabet
        if (isalpha(text[i]) != 0)
        {
            letters++;
        }
    }
    return letters;
}


int count_words(string text)
{
    int words = 0;
    // Loop through string until null character
    for (int i = 0; text[i] != '\0'; i++)
    {
        // reach space = mark a word
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return ++words;
}


int count_sentences(string text)
{
    int sentences = 0;
    // Loop through string until null character
    for (int i = 0 ; text[i] != '\0' ; i++)
    {
        // if '.', '!' or '?' so end of a sentence
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

int calc(int letters, int words, int sentences)
{
    float l = letters / (float)words * 100;
    float s = sentences / (float)words * 100;

    int c = round(0.0588 * l - 0.296 * s - 15.8);
    return c;
}

File: readability.py
from cs50 import get_string


def is_word_end(c):
    return c == " "


def is_sentence_end(c):
    return c == "." or c == "!" or c == "?"


def is_letter(c):
    return c.isalpha()


def coleman_liau_index(letter_count, word_count, sentence_count):
    l = 100 * letter_count / word_count
    s = 100 * sentence_count / word_count
    result = (0.0588 * l) - (0.296 * s) - 15.8
    return round(result)


# Input Text
text = get_string("Text: ")

count = 0
letter_count = 0
word_count = 0
sentence_count = 0

for c in text:
    if is_word_end(c):
        word_count += 1
    elif is_sentence_end(c):
        sentence_count += 1
    elif is_letter(c):
        letter_count += 1

# Added count of last word
word_count += 1

index = coleman_liau_index(letter_count, word_count, sentence_count)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(("Grade {}").format(index))
